src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx

```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
- */
-export class Backup extends Construct {
-  constructor(scope: Construct, id: string, props: BackupStackProps) {
-    super(scope, id, props);
+Custom constructs allow you to encapsulate common patterns into reusable components. This helps you implement best practices, accelerate development, and maintain consistency across applications.
 
-    const { database } = props;
-
-    // create the backup plan
-    const plan = new backup.BackupPlan(this, 'BackupPlan', {
-      backupPlanRules: [
-        new backup.BackupPlanRule({
-          completionWindow: cdk.Duration.hours(2),
-          startWindow: cdk.Duration.hours(1),
-          scheduleExpression: events.Schedule.cron({
-            day: '15',
-            hour: '3',
-            minute: '30',
-          }),
-          moveToColdStorageAfter: cdk.Duration.days(30),
-        }),
-      ],
+A common use case is creating a custom notification construct that combines a Lambda functions with SNS and SES.
+
+This AWS CDK construct implements a decoupled notification system using Amazon SNS and Lambda. It allows publishing notification messages to an SNS topic from one Lambda function, and processing those messages asynchronously using a separate Lambda subscribed to the topic.
+
+The key components are:
+
+- An Amazon SNS topic to receive notification messages
+- A Lambda function to publish messages to the SNS topic
+- A second Lambda subscribed to the topic that processes the messages and sends emails via Amazon SES
+
+The publisher Lambda allows publishing a message containing the email subject, body text, and recipient address. The emailer Lambda retrieves messages from the SNS topic and handles sending the actual emails.
+
+The `CustomNotifications` custom CDK construct can be defined as follows:
+
+```ts title="amplify/custom/CustomNotifications/resource.ts"
+import { Runtime } from "aws-cdk-lib/aws-lambda";
+import {
+  NodejsFunction,
+  NodejsFunctionProps,
+} from "aws-cdk-lib/aws-lambda-nodejs";
+import * as sns from "aws-cdk-lib/aws-sns";
+import * as subs from "aws-cdk-lib/aws-sns-subscriptions";
+import { Construct } from "constructs";
+import { dirname, join } from "path";
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+export type Message = {
+  subject: string;
+  body: string;
+  recipient: string;
+};
+interface CustomNotificationsProps {
+  sourceAddress: string;
+}
+
+export class CustomNotifications extends Construct {
+  constructor(scope: Construct, id: string, props: CustomNotificationsProps) {
+    super(scope, id);
+
+    const { sourceAddress } = props;
+
+    // Create SNS topic
+    const topic = new sns.Topic(this, "NotificationTopic");
+
+    const nodeJsFunctionProps: NodejsFunctionProps = {
+      bundling: {
+        externalModules: [
+          "aws-sdk", // Use the 'aws-sdk' available in the Lambda runtime
```

Comment:
The Node 18 runtime comes with AWS SDK v3, so it should be safe to remove this

Reply:
I needed this when deploying it in my sandbox, so leaving it for now.